File: src/main/java/com/leesh/inflpick/influencer/domain/InfluencerId.java
package com.leesh.inflpick.influencer.domain;

import com.leesh.inflpick.common.port.out.UuidHolder;
import lombok.NonNull;

public record InfluencerId(@NonNull String uuid) {

    public InfluencerId(@NonNull UuidHolder uuidHolder) {
        this(uuidHolder.uuid());
        if (this.uuid.isBlank()) {
            throw new IllegalArgumentException("uuid must not be blank");
        }
    }
}

File: src/main/kotlin/com/leesh/inflpick/influencer/domin/InfluencerId.kt
package com.leesh.inflpick.influencer.domin

import com.leesh.inflpick.common.domain.DomainId

class InfluencerId(private val id: String): DomainId {
    override fun isSaved(): Boolean {
        return id.isNotEmpty()
    }

    override fun toString(): String {
        return id
    }

    companion object {
        fun fromString(id: String): InfluencerId {
            return InfluencerId(id)
        }
    }
}